### ISSUE TITLE
type-literal-delimiter should suggest comma instead of semicolon

### ISSUE THREAD

**Issue by phaux** [palantir/tslint#4196]
### Rule Suggestion

**Is your rule for a general problem or is it specific to your development style?**

Object type literals in typescript allow comma, semicolon or newline separators between properties, **but** object literals in JS use comma. I just noticed a discrepency here.

```ts
// this is valid typescript
type Foo = {
  a: any, b: any; c: any
  d: any
}
```

**What does your suggested rule do?**

The rule `type-literal-delimiter` requires semicolons as separators.

**List several examples where your rule could be used**

- common sense would suggest that only commas are allowed
- flow allows only commas afaik

**Additional context**

TSLint suggests to use semicolons as separators and there's no option to set comma or newline as preferred separator. It even suggests semicolons when the rule `semicolon` is set to `"never"`, unlike in the case of interface declaration.

I think that at least the default should be to use comma, as that is the default object syntax.

**Comment by JoshuaKGoldberg** [comment]
@phaux this would be a pretty straightforward change to make, but is there much of a demand for it to justify the added maintenance cost? Are there any projects around that prefer the comma style?

It'd also be useful to see here why semicolons were preferred here by the community in the first place _(was it happenstance? a conscious decision?)_, if there's a record of it.

**Comment by goodmind** [comment]
Any updates?

**Comment by goodmind** [comment]
No idea, but I would want to enforce commas because I'm migrating to Flow
